src/FormMigration/Steps/FormFieldManager.php

```diff
@@ -0,0 +1,352 @@
+
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/file-upload',
+            'attributes' => [
+                'maxFileSize' => $field['max_size'],
+                'allowedFileTypes' => $allowedFileTypes,
+            ]
+        ]);
+    }
+
+    private function addHiddenField($field): BlockModel
+    {
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/hidden',
+        ]);
+    }
+
+    private function addHtmlField($field): BlockModel
+    {
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/html',
+            'attributes' => [
+                'htmlCode' => $field['html'],
+            ]
+        ]);
+    }
+
+    private function addMultiSelectField($field): BlockModel
+    {
+        $fieldType = $field['template'] === 'checkbox_field' ? 'checkbox' : 'dropdown';
+        $options = array_map(function ($option) use ($field) {
+            return [
+                'label' => $option,
+                'value' => '',
+                'checked' => in_array($option, $field['selected'], true),
+            ];
+        }, array_filter($field['options']));
+
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/multi-select',
+            'attributes' => [
+                'fieldType' => $fieldType,
+                'options' => $options,
+            ]
+        ]);
+    }
+
+    private function addPhoneField($field): BlockModel
+    {
+        $phoneFormat = $field['format'] === 'domestic' ? 'domestic' : 'unformatted';
+
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/phone',
+            'attributes' => [
+                'phoneFormat' => $phoneFormat,
+            ]
+        ]);
+    }
+
+    private function addRadioField($field): BlockModel
+    {
+        $options = array_map(function ($option) use ($field) {
+            return [
+                'label' => $option,
+                'value' => '',
+                'checked' => $option === $field['selected'],
+            ];
+        }, array_filter($field['options']));
+
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/radio',
+            'attributes' => [
+                'options' => $options,
+            ]
+        ]);
+    }
+
+    private function addTextField($field): BlockModel
+    {
+        return BlockModel::make([
+            'name' => 'givewp/text',
+        ]);
+    }
+
+    private function addTextareaField($field): BlockModel
+    {
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/textarea',
+            'attributes' => [
+                'rows' => $field['rows'],
+            ],
+        ]);
+    }
+
+    private function addUrlField($field): BlockModel
+    {
+        return BlockModel::make([
+            'name' => 'givewp-form-field-manager/url',
+        ]);
+    }
+
+    private function addSection($field): void
```

Comment:
Multiple sections are inserted in reverse order.

| V2 | V3 |
| --- | --- |
| ![image](https://github.com/impress-org/givewp/assets/10858303/cf015f87-85aa-4daa-a949-0da8c7d2c136) | ![image](https://github.com/impress-org/givewp/assets/10858303/860ead38-b43d-4444-b9bc-43166bcfb829) |

Reply:
Oh! Found the issue. Since the `insertAfter` uses a string, we are passing `givewp/section` to it but not different index so all new sections are being inserted right after the first `givewp/section`. I'll work to get the index of the current section and so pass that info to the `insertAfter` method.